### PR TITLE
[ESP32] SecureCertDACProvider implementation to read attestation info from esp_secur…

### PR DESCRIPTION
**Pull request by shripad621git**
**Change Overview**
 Added the code related to SecureCertDACProvider to read attestation info from esp_secure_cert_partition.
 Added the configuration of SEC_CERT_DAC_PROVIDER and code which helps to decide which implementation of
 DAC Provider to use.
 Added esp_secure_cert_mgr component.

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: config/esp32/components/chip/CMakeLists.txt [hunk 1]
```diff
@@ -341,7 +346,10 @@ if((NOT CONFIG_USE_MINIMAL_MDNS) AND (CONFIG_ENABLE_WIFI_STATION OR CONFIG_ENABL
 endif()
 
 idf_component_get_property(main_lib main COMPONENT_LIB)
+idf_component_get_property(esp32_secure_cert_mgr_lib espressif__esp_secure_cert_mgr COMPONENT_LIB)
+
```
Comment (dhrishi) [inline config/esp32/components/chip/CMakeLists.txt:350]: Should this and the below line be added when `CONFIG_SEC_CERT_DAC_PROVIDER` is enabled?

File: config/esp32/components/chip/Kconfig [hunk 2]
```diff
@@ -678,6 +678,12 @@ menu "CHIP Device Layer"
             help
                 Enable ESP32 Device LocationCapability
 
+        config SEC_CERT_DAC_PROVIDER
+            bool "Use Secure Cert DAC Provider"
+            default n
+            help
+                Enable Secure Cert DAC Provider
```
Comment (dhrishi) [inline config/esp32/components/chip/Kconfig:685]: Please provide additional information about from where the certs are read

File: src/platform/ESP32/ESP32SecureCertDACProvider.cpp [hunk 3]
```diff
@@ -0,0 +1,148 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
```
Comment (dhrishi) [inline src/platform/ESP32/ESP32SecureCertDACProvider.cpp:3]: 2023?

File: src/platform/ESP32/ESP32SecureCertDACProvider.h [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
```
Comment (dhrishi) [inline src/platform/ESP32/ESP32SecureCertDACProvider.h:3]: 2023

File: src/platform/ESP32/ESP32SecureCertDACProvider.h [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+{
+public:
+    ESP32SecureCertDACProvider() : Credentials::DeviceAttestationCredentialsProvider() {}
+    ~ESP32SecureCertDACProvider() {}
+    CHIP_ERROR GetCertificationDeclaration(MutableByteSpan & outBuffer) override;
+    CHIP_ERROR GetFirmwareInformation(MutableByteSpan & out_firmware_info_buffer) override;
+    CHIP_ERROR GetDeviceAttestationCert(MutableByteSpan & outBuffer) override;
+    CHIP_ERROR GetProductAttestationIntermediateCert(MutableByteSpan & outBuffer) override;
+    CHIP_ERROR SignWithDeviceAttestationKey(const ByteSpan & messageToSign, MutableByteSpan & outSignBuffer) override;
+};
```
Comment (dhrishi) [inline src/platform/ESP32/ESP32SecureCertDACProvider.h:34]: Can we also add some documentation in `docs/guides/esp32` about using the esp_secure_cert partition for the attestation certificates? And also, the other options (factory data provider etc.) that need to be enabled while using this option

**Review by shubhamdp — APPROVED** [review]

File: src/platform/ESP32/ESP32SecureCertDACProvider.cpp (hunk 3 above)
Comment (shripad621git) [reply src/platform/ESP32/ESP32SecureCertDACProvider.cpp]: Made the suggested change.

File: src/platform/ESP32/ESP32SecureCertDACProvider.h (hunk 4 above)
Comment (shripad621git) [reply src/platform/ESP32/ESP32SecureCertDACProvider.h]: Made the suggested change.

File: config/esp32/components/chip/Kconfig (hunk 2 above)
Comment (shripad621git) [reply config/esp32/components/chip/Kconfig]: The suggested change is made.

**Review by dhrishi — APPROVED** [review]

**Pull request merged**